### DISCUSSION TITLE
Sagernet over Mobile Hotspot

### DISCUSSION THREAD

**Discussion by shahramdll** [SagerNet/SagerNet#494] [Q&A]
Hello,
Can I use SagerNet over Mobile Hotspot?
I used VPN Hotspot app that it run a vpn over hotspot with root permission and I shared SagerNet over hotspot successfully but the problem is that I need to apply Sagernet Routes also to hotspot and I couldn't find any solution or option to do that in Sagernet nor VPN Hotspot app, 
Any help?

**Comment by iKirby** [comment]
Routing is handled by v2ray-core, so when you share the VPN connection, the routes are still effective. Apps VPN mode is handled by Android system and only applies to the current device.

**Reply by shahramdll** [reply]
Routing is ok in my device and works fine, but not applied to other device that connect through my hotspot, Is there anyway to apply routing to other devices that connected to mobile hotspot with sagernet vpn?

**Reply by iKirby** [reply]
I think you misunderstood how routing works. When you access the internet, traffic goes to the VPN interface to be handled by the VPN app.

```
App --- VPN interface --- SagerNet --- the internet
```

When you share your VPN connection, traffic also goes through VPN interface.

```
Devices connected to the hotspot --- Hotspot --- VPN interface --- SagerNet --- the internet
```

Routing is handled by SagerNet's v2ray-core on the hotspot device, so it should be the same unless you use other proxy apps on devices connected to the hotspot.

I tested on my phone running LineageOS, it has the option "Allow clients to use VPNs". After connecting another device to the hotspot, I can confirm that routing is exactly the same as on my phone. VPNHotspot should work the same way if I'm not wrong, I can test it later on a rooted device when I'm home.

**Reply by iKirby** [reply]
I tested sharing VPN connection on a rooted device using VPNHotspot, and can confirm that it is working as expected. So I think in your case it is already working, you can use some sites to check IP address and see if routing works. The key point is that routing is handled by SagerNet, and there is no need to manage routes on devices connected to the hotspot.

**Comment by shahramdll** [comment]
Unfortunately there is not any option in my Samsung device running original Rom by default to share device VPN and when I connect my Samsung smart TV to my phone hotspot with sagernet routing and VPN hotspot app, sagernet routing not apply to my connected device, I checked my IP on many sites that they are in my bypass list and I see my VPN IP not my local IP.
I've a Samsung tablet with lineage custom Rom, I'll check on it later and tell the result.

**Comment by cross-hello** [comment]
@shahramdll  You should allow **allow Connections from the LAN** in SagerNet, then you will get two port allowing to be connectd, one for *http/https*, and other for *socks*. 
When your client connect to the hostpot shared via Samsung phone running SagerNet with above option, specify the **Proxy setting** in **Wifi** configuration. Then fill the port inside bound.
![thumb-Screenshot_2022_1031_170513](https://user-images.githubusercontent.com/42733664/198972060-5b4ee547-9cfc-4469-8e69-dfa760766839.png)
